packages/node/src/integrations/tracing/nest/sentry-nest-instrumentation.ts

```diff
@@ -163,33 +170,67 @@ export class SentryNestInstrumentation extends InstrumentationBase {
 
             target.prototype.intercept = new Proxy(target.prototype.intercept, {
               apply: (originalIntercept, thisArgIntercept, argsIntercept) => {
-                const [executionContext, next, args] = argsIntercept;
+                const context: MinimalNestJsExecutionContext = argsIntercept[0];
+                const next: CallHandler = argsIntercept[1];
+                const request = context.switchToHttp().getRequest();
+
                 const prevSpan = getActiveSpan();
+                let afterSpan: Span;
 
                 return startSpanManual(getMiddlewareSpanOptions(target), (span: Span) => {
-                  const nextProxy = new Proxy(next, {
-                    get: (thisArgNext, property, receiver) => {
-                      if (property === 'handle') {
-                        const originalHandle = Reflect.get(thisArgNext, property, receiver);
-                        // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                        return (...args: any[]) => {
-                          span.end();
-
-                          if (prevSpan) {
-                            return withActiveSpan(prevSpan, () => {
-                              return Reflect.apply(originalHandle, thisArgNext, args);
-                            });
-                          } else {
-                            return Reflect.apply(originalHandle, thisArgNext, args);
+                  // eslint-disable-next-line @typescript-eslint/unbound-method
+                  next.handle = new Proxy(next.handle, {
+                    apply: (originalHandle, thisArgHandle, argsHandle) => {
+                      span.end();
+
+                      if (prevSpan) {
+                        return withActiveSpan(prevSpan, () => {
+                          const handleReturnObservable = Reflect.apply(originalHandle, thisArgHandle, argsHandle);
+                          if (request.AFTER_ROUTE_SPAN_KEY) {
+                            return handleReturnObservable;
                           }
-                        };
-                      }
 
-                      return Reflect.get(target, property, receiver);
+                          afterSpan = startInactiveSpan(getMiddlewareSpanOptions(target, 'Interceptor - After Route'));
+                          return handleReturnObservable;
+                        });
+                      } else {
+                        const handleReturnObservable = Reflect.apply(originalHandle, thisArgHandle, argsHandle);
+
+                        if (request.AFTER_ROUTE_SPAN_KEY) {
+                          return handleReturnObservable;
+                        }
+
+                        afterSpan = startInactiveSpan(getMiddlewareSpanOptions(target, 'Interceptor - After Route'));
+                        return handleReturnObservable;
```

Comment:
```suggestion
                        if (!request.AFTER_ROUTE_SPAN_KEY) {
                          afterSpan = startInactiveSpan(getMiddlewareSpanOptions(target, 'Interceptor - After Route'));
                        }

                        return handleReturnObservable;
```

l: I find this way more readable